### PR TITLE
Implement javascript plugins

### PR DESCRIPTION
**Pull request by 3stadt**

### CODE REVIEW

File: src/queue/worker.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+func commandWorker(job structs.Job, connection *irc.Connection) {
+	var msg *structs.Message
+	var err error
+	if _, ok := handlers.CommandMap[job.Command]; ok {
+		err = handlers.CommandMap[job.Command](job.Channel, job.Sender, job.Params, connection)
+		if err != nil {
+			fmt.Println(err)
+		}
+	} else{
```
Comment (leichteckig) [inline src/queue/worker.go]: I know - peanut - but could you add a whitespace after the 'else'?

File: cmd/GoTBot/main.go [hunk 2]
```diff
@@ -119,6 +141,36 @@ func main() {
...
+					continue
+				}
+				var commands structs.Commands
+				if _, err := toml.Decode(string(tomlData), &commands); err != nil {
+					log.Fatal(err)
+				}
+				for _, c := range commands.Command {
+					if _, ok := handlers.PluginCommandMap[c.Name]; ok {
+						handlers.PluginCommandMap[c.Name] = append(handlers.PluginCommandMap[c.Name], "./custom/plugins/" + file.Name() + "/" + c.EntryScript)
+					} else{
```
Comment (leichteckig) [inline cmd/GoTBot/main.go]: Here, there might be a whitespace missing. Please correct me if I'm wrong, though. :)

File: custom/plugins/type_hint_helper.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+/**
+ *
+ * This file is used to provide type hinting in your javascript plugins.
+ * It is as well a documentation which functions/global vars you can use.
+ * It is not used in any other way and is never called.
+ *
+ */
```
Comment (leichteckig) [inline custom/plugins/type_hint_helper.js]: Compliment for this file, the information in it is very helpful! :)

File: src/context/context.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import "github.com/asdine/storm"
+
+var Conf map[string]string
+var DB *storm.DB
+var PluginDB *storm.DB
+var Users storm.Node
+
+const CommandQueueName = "commands"
+const UserBucketName = "Users"
```
Comment (leichteckig) [inline src/context/context.go]: Why are you changing sentence case here?

File: src/db/db.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	user.LastActive = &now
+	SetUser(user)
+	return nil
+}
+
+func CreateUser(nick string) error {
+	now := time.Now()
+	newUser := structs.User{}
+	newUser.Name = nick
+	newUser.MessageCount = 1
```
Comment (leichteckig) [inline src/db/db.go]: Here you assume that the user has already written one message, even if he didn't. What would be the downside of using zero as value?

**Comment by 3stadt** [comment]
I'll do a new PR with the requested changes

**Pull request closed without merging**